PluginManager/PluginManager.cpp

```diff
@@ -277,26 +276,43 @@ namespace plugin
         return ResolvePathInternal(filename, "PLUGIN_PATH", "plugins/");
     }
 
-    std::filesystem::path PluginManager::ResolvePathInternal(std::string filename,
+    std::filesystem::path PluginManager::ResolvePathInternal(std::filesystem::path filename,
                                                const char* envVarOverride,
                                                std::string defaultPrefix)
     {
         if(filename.empty()) {
             throw std::logic_error{ "Received blank filename in ResolvePathInternal" };
         }
 
-        std::ifstream file(filename.c_str());
-
         // Respect absolute paths. We assume that client passed an absolute
         // path because they know the exact location which is fixed.
-        if (file.is_open() || filename[0] == '/') {
+        if(std::filesystem::exists(filename) || filename.is_absolute()) {
```

Comment:
Check for absolute using std::filesystem rather than assuming a leading forward slash.